addons/point_of_sale/static/src/app/utils/pretty_console_log.js

```diff
@@ -13,4 +18,23 @@ export function logPosMessage(type, functionName, message, color = "#A1A1A1", ar
         console.trace("Call stack:");
         console.groupEnd();
     }
+    const timestamp = luxon.DateTime.now().toUTC().toFormat("yyyy-LL-dd HH:mm:ss");
+    posLogger.log(`${timestamp} [${type}]: ${functionName} - ${message}`);
+    if (args.length) {
+        try {
+            posLogger.log(`args: ${JSON.stringify(args)}`);
+        } catch {
+            // In case the args are not serializable
+            posLogger.log(`args: ${args}`);
+        }
+    }
+}
+
+export async function downloadPosLogs() {
+    const logs = await posLogger.getLogs();
+    const blob = new Blob([logs.join("\n")], {
+        type: "text/plain",
+    });
+    const filename = `pos_logs_${luxon.DateTime.now().toUTC().toFormat("yyyy-LL-dd-HH-mm-ss")}.txt`;
```

Comment:
Why not exporting it as json, it would be more readable for the args, wdyt ?

```suggestion
    logs.forEach((l) => {
        if (l["args"]) {
            l["args"] = JSON.parse(l["args"]);
        }
    });
    const blob = new Blob([JSON.stringify(logs, null, 2)], {
        type: "application/json",
    });
    const filename = `pos_logs_${luxon.DateTime.now()
        .toUTC()
        .toFormat("yyyy-LL-dd-HH-mm-ss")}.json`;
```